### DISCUSSION TITLE
Running two seperate exe generated from an Air bundle in the same directory

### DISCUSSION THREAD

**Discussion by Kelsonk** [airsdk/Adobe-Runtime-Support#2590] [General]
Hi everyone, I have a really specific problem that's driving me a bit mad. I've made a Steam game a few years ago with Adobe Air, which all runs perfectly fine. It uses the bundle target, so all game assets are included in the same folder (the initial content swf, the executable, all art assets, and other folders like META-INF (which contains the application xml), and the packaged AIR folder.)

I wanted to make a little fun screensaver that runs in the same game directory, and uses all the same game assets in that directory.

The problem is that the screensaver needs to load a different swf (the initial content part of the application xml needs to be changed). As far as I can tell, the executables only run the initial content provided in the application xml in the META_INF folder.

I can certainly make it so that the screensaver can be its own app, creating a separate folder for the exe and all the game assets. But this would be a bummer, as I just wanted to include a small file to run a separate program.

So I guess my question is, can I run two executables in the same folder? Can I point my second executable to load a different application xml or run different initial content?

My current game folder looks like this:

/assets
/game.exe
/game.swf
/META_INF (which says that the initial content is game.swf in the application xml)
/AIR
/game2.exe (I need this to run game2.swf, determined from the application xml)
/game2.swf

**Comment by Kelsonk** [comment]
I made a solution but it was pretty goofy. Doesn't feel quite right but it works.

/game.exe
/game.swf
/META_INF (which says that the initial content is game.swf in the application xml)
/AIR
/game2/game2.exe (I need this to run game2.swf, determined from the application xml)
/game2/game2.swf
/game2/META_INF
/game2/AIR
/game2/assets

**Reply by MalacTheLittle** [reply]
This isn't quite a solution since exe files are in different folders :)

**Reply by Kelsonk** [reply]
> This isn't quite a solution since exe files are in different folders :)

Yeah but at least I don't have to have 300 mbs of assets install twice (but unfortunately I still must have 2 copies of Air, one in each folder.)

**Reply by ajwfrost** [reply]
The .exe files are hard-wired to load a file "META-INF/AIR/application.xml" and this then gives the initial SWF content, so there's no real way around this part. But could your initial SWF be a tiny bootstrap that decides whether to then load in 'game.swf' or 'game2'swf'?
